src/Faker/Provider/Base.php

```diff
@@ -546,7 +546,7 @@ public function optional($weight = 0.5, $default = null)
     {
         // old system based on 0.1 <= $weight <= 0.9
         // TODO: remove in v2
-        if ($weight > 0 && $weight < 1 && mt_rand() / mt_getrandmax() <= $weight) {
+        if ($weight > 0 && $weight <= 1 && mt_rand() / mt_getrandmax() <= $weight) {
```

Comment:
Why do we want an equals here? If the weight is 1 it will always be the same value so not optional?